test/phpunit/SecurityTest.php

```diff
@@ -224,62 +225,71 @@ public function testSqlAndScriptInjectWithPHPUnit()
-		$result=testSqlAndScriptInject($test, 0);
+		$test = "<img src=x one<a>rror=alert(document.location)";
+		$result = testSqlAndScriptInject($test, 0);
 		$this->assertGreaterThanOrEqual($expectedresult, $result, 'Error on testSqlAndScriptInject kkk');
 
-		$test="<a onpointerdown=alert(document.domain)>XSS</a>";
-		$result=testSqlAndScriptInject($test, 0);
+		$test = "<a onpointerdown=alert(document.domain)>XSS</a>";
+		$result = testSqlAndScriptInject($test, 0);
 		$this->assertGreaterThanOrEqual($expectedresult, $result, 'Error on testSqlAndScriptInject lll');
 
-		$test='<a onscrollend=alert(1) style="display:block;overflow:auto;border:1px+dashed;width:500px;height:100px;"><br><br><br><br><br><span+id=x>test</span></a>';	// Add the char %F6 into the variable
-		$result=testSqlAndScriptInject($test, 0);
+		$test = '<a onscrollend=alert(1) style="display:block;overflow:auto;border:1px+dashed;width:500px;height:100px;"><br><br><br><br><br><span+id=x>test</span></a>';	// Add the char %F6 into the variable
+		$result = testSqlAndScriptInject($test, 0);
 		//print "test=".$test." result=".$result."\n";
 		$this->assertGreaterThanOrEqual($expectedresult, $result, 'Error on testSqlAndScriptInject mmm');
 
 
-		$test="Text with ' encoded with the numeric html entity converted into text entity &#39; (like when submitted by CKEditor)";
-		$result=testSqlAndScriptInject($test, 0);	// result must be 0
+		$test = "Text with ' encoded with the numeric html entity converted into text entity &#39; (like when submitted by CKEditor)";
+		$result = testSqlAndScriptInject($test, 0);	// result must be 0
 		$this->assertEquals(0, $result, 'Error on testSqlAndScriptInject mmm, result should be 0 and is not');
 
-		$test ='<a href="j&Tab;a&Tab;v&Tab;asc&NewLine;ri&Tab;pt:&lpar;a&Tab;l&Tab;e&Tab;r&Tab;t&Tab;(document.cookie)&rpar;">XSS</a>';
-		$result=testSqlAndScriptInject($test, 0);
+		$test = '<a href="j&Tab;a&Tab;v&Tab;asc&NewLine;ri&Tab;pt:&lpar;a&Tab;l&Tab;e&Tab;r&Tab;t&Tab;(document.cookie)&rpar;">XSS</a>';
+		$result = testSqlAndScriptInject($test, 0);
 		$this->assertGreaterThanOrEqual($expectedresult, $result, 'Error on testSqlAndScriptInject nnn, result should be >= 1 and is not');
 
-		$test="/dolibarr/htdocs/index.php/".chr('246')."abc";	// Add the char %F6 into the variable
-		$result=testSqlAndScriptInject($test, 2);
+		$test = "/dolibarr/htdocs/index.php/".chr('246')."abc";	// Add the char %F6 into the variable
+		$result = testSqlAndScriptInject($test, 2);
 		//print "test=".$test." result=".$result."\n";
 		$this->assertGreaterThanOrEqual($expectedresult, $result, 'Error on testSqlAndScriptInject with a non valid UTF8 char');
+
+		$vars = array("PHP_SELF" => '/c/phptools/bin/phpunit');
+		$result = analyseVarsForSqlAndScriptsInjection($vars, 2);
+		$this->assertTrue($result, "Error on testSqlAndScriptInject for {$vars['PHP_SELF']} that should be accepted");
+
+		$vars = array("PHP_SELF" => 'c:\tools\bin\phpunit');
```

Comment:
According to PHP doc PHP_SELF should be the relative path from the web server. 
So having a content like c:\tools\bin\phpunit is not an expected content.
What value do you have for $_SERVER["PHP_SELF"]  ?

Reply:
That is exactly the value I have.

Reply:
> That is exactly the value I have.

Looks strange. Which webserver are you using to have this value ?

Reply:
No webserver - just phpunit.
See #28187 https://github.com/Dolibarr/dolibarr/actions/runs/7969576072 .
And now also #28301 https://github.com/Dolibarr/dolibarr/actions/runs/7969600927 .